src/routes/webhooks/paddle.ts

```diff
@@ -405,6 +370,66 @@ const notifyNewPaddleTransaction = async ({
   await webhooks.transactions.send({ blocks });
 };
 
+export const processGiftedPayment = async (
+  data: TransactionCompletedEvent['data'],
+) => {
+  const con = await createOrGetConnection();
+  const { gifter_id, user_id } = data.customData as PaddleCustomData;
+
+  if (user_id === gifter_id) {
+    logger.error({ type: 'paddle', data }, 'User and gifter are the same');
+    return;
+  }
+
+  const gifterUser = await con.getRepository(User).findOneBy({ id: gifter_id });
+
+  if (!gifterUser) {
+    logger.error({ type: 'paddle', data }, 'Gifter user not found');
+    return;
+  }
+
+  const targetUser = await con.getRepository(User).findOne({
+    select: ['subscriptionFlags'],
+    where: { id: user_id },
+  });
+
+  if (isPlusMember(targetUser?.subscriptionFlags?.cycle)) {
+    logger.error({ type: 'paddle', data }, 'User is already a Plus member');
+    return;
+  }
+
+  await con.getRepository(User).update(
+    {
+      id: user_id,
+    },
+    {
+      subscriptionFlags: updateSubscriptionFlags({
+        cycle: SubscriptionCycles.Yearly,
+        createdAt: data?.createdAt,
+        subscriptionId: data?.id,
+        gifterId: gifter_id,
+        giftExpirationDate: addMilliseconds(
+          new Date(),
+          plusGiftDuration,
+        ).toISOString(),
+      }),
+      flags: updateFlagsStatement({ showPlusGift: true }),
+    },
+  );
+};
+
+export const processTransactionCompleted = async (
+  result: TransactionCompletedEvent,
+) => {
```

Comment:
Let's make it accept object `{ data }` or whatever, easier to expand later

Reply:
The whole object was needed originally, and didn't want to make any change outside the gifting, but yeah, I can update the other functions too.

Reply:
you can still do the whole object, just wanted to move to object syntax

Reply:
```ts
export const processTransactionCompleted = async ({
  result: TransactionCompletedEvent,
}) => Promise

Reply:
Ah, cool, yeah, we can do that.